src/librustc/mir/interpret/error.rs

```diff
@@ -150,28 +154,32 @@ impl<'tcx> ConstEvalErr<'tcx> {
             err => err.to_string(),
         };
 
-        let mut err = if let (Some(lint_root), false) = (lint_root, must_error) {
-            let hir_id = self
-                .stacktrace
-                .iter()
-                .rev()
-                .filter_map(|frame| frame.lint_root)
-                .next()
-                .unwrap_or(lint_root);
-            tcx.struct_span_lint_hir(
-                rustc_session::lint::builtin::CONST_ERR,
-                hir_id,
-                tcx.span,
-                message,
-            )
-        } else if must_error {
+        // FIXME(RalfJ): Why do we "must_error" on `SizeOverflow` and on nothing else?
```

Comment:
This was introduced in https://github.com/rust-lang/rust/pull/63152 otherwise you could end up with a broken promoted if you allowed this lint. Without promotion it would have been a hard error, too.